staging/src/k8s.io/component-helpers/apimachinery/lease/controller.go

```diff
@@ -65,9 +65,8 @@ type controller struct {
 	latestLease *coordinationv1.Lease
 
 	// newLeasePostProcessFunc allows customizing a lease object (e.g. setting OwnerReference)
-	// before every time the lease is created/refreshed(updated). Note that an error will block
-	// a lease CREATE, causing the controller to retry next time, but an error won't block a
-	// lease UPDATE.
+	// before every time the lease is created/refreshed(updated).
+	// Note that an error will block the lease operation.
 	newLeasePostProcessFunc ProcessLeaseFunc
```

Comment:
Need to make sure changing this assumption is okay.

I only see two usages of this controller, using these `newLeasePostProcessFunc`-s:
1. `kube-apiserver`:https://github.com/kubernetes/kubernetes/blob/master/pkg/controlplane/instance.go#L653-L682
    - This never returns an `error`, so the change is fine.
2. `kubelet`: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/util/nodelease.go#L30-L54

Reply:
What about outside of kube itself? https://grep.app/search?q=coordinationv1.Lease

Reply:
current behavior will be always introducing a buggy behavior, no?

Reply:
@sttts we'd need to look for usages of this controller, not use `Lease`: https://grep.app/search?q=k8s.io/component-helpers/apimachinery/lease

I only see 1 usage (karmada-io/karmada) that isn't k8s or vendored-k8s. It's using this `newLeasePostProcessFunc`, which has the same behavior as kubelet's (and so would probably encounter the same bug 😄 ): https://github.com/karmada-io/karmada/blob/e5277b6317ac1a4717f5fac4057caf51a5d248fc/pkg/util/clusterlease.go#L16-L37

Reply:
This library *does* have the standard disclaimer that it has no compatibility guarantees, it in direct support of Kubernetes, etc: https://github.com/kubernetes/component-helpers#compatibility

If we're concerned about the subtlety, I _could_ make a breaking cosmetic change in the name or something to flag this at compile time? But I think that'd be more annoying than helpful.